src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            resize();
+        }
+        int index = hash(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+            return;
+        }
+        checkKey(newNode);
     }
 
     @Override
     public V getValue(K key) {
+        int index = hash(key);
+        Node<K, V> getNode = table[index];
+        while (getNode != null) {
+            if (Objects.equals(getNode.key, key)) {
+                return getNode.value;
+            }
+            getNode = getNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public void resize() {
+        size = 0;
+        int newThreshold = threshold * 2;
+        int newCapacity = table.length * 2;
+        threshold = newThreshold;
```

Comment:
magic numbers must be constant. Also, what is threshold? From what variable it is calculated?